lib/logging.dart

```diff
@@ -64,7 +64,7 @@ class Logger {
   /// It can be useful when you just need a local short-living logger,
   /// which you'd like to be garbage-collected later.
   factory Logger.detached(String name) {
-    return new Logger._internal(name, null, new Map<String, Logger>());
+    return new Logger._internal(name, null!, new Map<String, Logger>());
```

Comment:
Note that any time the migration tool generates `null!`, as it has now, it should issue a big warning to the user, because the expression `null!` is guaranteed to fail at runtime.  See https://github.com/dart-lang/sdk/issues/38345.

In this particular case, we're generating `null!` due to a bug (https://github.com/dart-lang/sdk/issues/38341).  What's happening is that in `logging_test.dart`, we have the code:
```dart
test('cannot directly manipulate Logger.children', () {
  var loggerAB = new Logger('a.b');
  var loggerA = loggerAB.parent;

  expect(loggerA.children['b'], same(loggerAB), reason: 'can read Children');
  ...
}
```
The use of `loggerA` in the `expect` call will fail if `loggerA` is `null`, and that use post-dominates the declaration of `loggerA`, so the migration engine concludes that `loggerA` should have a non-nullable type.  Since the type of `loggerA` is inferred from the type of the field `Logger.parent` (line 39), the engine concludes that `Logger.parent` should be non-nullable, and thus the attempt to initialize it with a nullable value here needs to be null-checked.

This is another case where it would have been helpful to have some diagnostic output from the migration tool, since there were several nullability nodes and edges involved:
- This line of code caused an edge from `always` to a node called `inferred` (representing the nullability of the inferred type of `Logger._internal`'s `this.parent` argument).
- There was a "union" edge from `inferred` to a node called `type(1258)` (representing the nullability of the field `Logger.parent`).  This edge was caused by the fact that `this.` parameters have the same type as their corresponding fields.
- There was a further "union" edge from `type(1258)` to a node called `inferred_71` (representing the nullability of the variable `loggerA`). This was the incorrect edge.
- There was an edge from `inferred_71` to `never`, caused by the use of `loggerA` in `loggerA.children`.

I believe that once https://github.com/dart-lang/sdk/issues/38341 is fixed, `Logger.parent` will be considered nullable, so this `!` won't appear here.